.github/workflows/ci.yml

```diff
@@ -229,14 +229,56 @@ jobs:
         tesseract --version
         make test-extra-${{ matrix.extra }} CI=true
 
+  setup_ingest:
+    strategy:
+      matrix:
+        python-version: [ "3.8","3.9","3.10","3.11" ]
+    runs-on: ubuntu-latest
+    env:
+      NLTK_DATA: ${{ github.workspace }}/nltk_data
+    needs: [setup]
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/cache/restore@v3
+        id: base-virtualenv-cache
+        with:
+          path: |
+            .venv
+            nltk_data
+          key: unstructured-${{ runner.os }}-${{ matrix.python-version }}-${{ env.GHA_CACHE_KEY_VERSION }}-${{ hashFiles('requirements/*.txt') }}
+      # Due to the dependency on setup, the cache should exist before this is ran. Set failed if it wasn't found.
```

Comment:
given that we're WAY over the [10gb cache limit](https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#usage-limits-and-eviction-policy). we generally try to do the right thing with a setup job and leveraging that in subsequent jobs, but I suspect at any point these can be ejected (which is probably a good reason for some wonkiness), so I believe that's why when it's not found IN the job, we typically just go ahead and rebuild. e.g. like we do [here](https://github.com/Unstructured-IO/unstructured/blob/main/.github/workflows/ci.yml#L35-L40).